### ISSUE TITLE
S1854 FP: variable used in finally block

### ISSUE THREAD

**Issue by glen-nicol** [SonarSource/sonar-dotnet#4085]
### Description

The finally block does not seem to be involved in the analysis of S1854. It advises the user to remove the assignment which leads to CS0165 (Use of unassigned local variable).

### Repro steps

```csharp
public class Program
{
    private enum ReproEnumType
    {
        Value1,
        Value2
    }

    public static void Main(string[] args)
    {
        ReproEnumType result = ReproEnumType.Value2;
        try
        {
            result = ReproEnumType.Value1;
        }
        finally
        {
            UseEnum(result);
        }
    }

    private static void UseEnum(ReproEnumType reproEnum)
    {
        _ = reproEnum;
    }
}
```

### Expected behavior

I expect the assignment requirements of the usages to be correctly analyzed and no warning to be issued.

### Actual behavior

False positive warning that if heeded produces broken code.

### Known workarounds

Suppress the warning

### Related information

* C# version: no langversion needed. So 7.3?
* SonarAnalyzer.CSharp 8.18.0.27296
* Visual Studio 2019 version: 16.8.4
* MSBuild / dotnet version: .0.102
* Operating System: Windows 10 2004 (I think)

**Comment by pavel-mikula-sonarsource** [comment]
Hi @glen-nicol,

Thank you for reporting this case, I can confirm it as False Positive. The root cause for this is already reported in #2600 that you can track so I'll close this one.
